### DISCUSSION TITLE
I deleted the "\Software\Microsoft\Windows\CurrentVersion\Lxss"

### DISCUSSION THREAD

**Discussion by sandyowlet** [microsoft/WSL#10501] [General] [answered]
### Windows Version

Microsoft Windows [version 10.0.19045.3324]

### WSL Version

2.0.0.0

### Are you using WSL 1 or WSL 2?

- [X] WSL 2
- [ ] WSL 1

### Distro Version

ubuntu

### Other Software

Linux version 5.15.123.1-microsoft-standard-WSL2 (root@5a891d5fa777) (gcc (GCC) 11.2.0, GNU ld (GNU Binutils) 2.37) #1 SMP Mon Aug 7 19:01:48 UTC 2023

### Repro Steps

I stupidly deleted the "\Software\Microsoft\Windows\CurrentVersion\Lxss", now I get "Class not registered, Error code: Wsl/CallMsi/REGDB_E_CLASSNOTREG".

I tried closing and reopening wsl.

Now I can access through the shortcut "wsl", but when entering "wsl" in cmd I will receive the above error prompt, and my vscode prompt cannot find the wsl distribution version.

and I stupidly didn't back up the registry.

### Expected Behavior

.

### Actual Behavior

.

**Comment by benhillis** [comment, marked as answer]
Could you try uninstall and re-installing the WSL MSI?

**Reply by sandyowlet** [reply]
My attempt to uninstall wsl2.0.0.0 kept failing yesterday, but another attempt today worked. Now my wsl is back to normal, thank you very much.

**Reply by pj185095** [reply]
I am facing the same issue but wsl --unistall is also giving the same error

**Comment by Eliav2** [comment]
this happens to me in the newset version of wsl.
details https://github.com/microsoft/WSL/issues/8268#issuecomment-1865083713
not sure why, but i cant run `wsl --update` because it break my wsl

**Comment by totszwai** [comment]
Anyone have a solution to this? WSL won't update, uninstall/reinstall doesn't do anything either.
